src/compile/mod.rs

```diff
@@ -1051,52 +1023,50 @@ fn codegen_programs(
             work_list.push((
                 func,
                 func_labels.clone(),
-                module.string_table.clone(),
                 global_vars.clone(),
                 module.name.clone(),
                 module.path.clone(),
             ));
         }
     }
 
+    work_list.swap(0, main_func_index);
```

Comment:
While I think I see how to do this, it is not clear to me how it would simplify the implementation. If you have a particular approach in mind would do this for this operation, I would appreciate if you were to implement it.